src/Http/Search/ProfileQuery.php

```diff
@@ -4,7 +4,7 @@
 
 namespace Werkspot\KvkApi\Http\Search;
 
-final class ProfileQuery implements QueryInterface
+class ProfileQuery implements QueryInterface
```

Comment:
The client developer can create a mock using the interface, so this class doesn't need to be not-final, for such a reason.

Reply:
Unfortunately this poses a problem because the client depends on this class signature directly:
https://github.com/Werkspot/kvk-api/blob/12ec13a81f6a2c54a6a3eab56a55d7ce318964dc/src/KvkClient.php#L32

I'd suggest introducing a new interface that extends the QueryInterface (ProfileQueryInterface) and type hinting on that in the client. Another option would be for the client to hint for QueryInterface instead.

Reply:
Indeed, the client should be depending on the `QueryInterface`, not on the `ProfileQuery` class.